controllers/nicclusterpolicy_controller_test.go

```diff
@@ -138,6 +138,84 @@ var _ = Describe("NicClusterPolicyReconciler Controller", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
+	Context("When MOFED precompiled tag does not exists", func() {
+		It("should set error message in status", func() {
+			By("Create Node")
+			node := &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+					Labels: map[string]string{
+						nodeinfo.NodeLabelMlnxNIC:       "true",
+						nodeinfo.NodeLabelOSName:        "ubuntu",
+						nodeinfo.NodeLabelCPUArch:       "amd64",
+						nodeinfo.NodeLabelKernelVerFull: "generic-9.0.1",
+						nodeinfo.NodeLabelOSVer:         "20.0.4"},
+					Annotations: make(map[string]string),
+				},
+			}
+			err := k8sClient.Create(context.TODO(), node)
+			Expect(err).NotTo(HaveOccurred())
+			By("Create NicClusterPolicy with MOFED ForcePrecompiled")
+			cr := mellanoxv1alpha1.NicClusterPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "nic-cluster-policy",
+					Namespace: "",
+				},
+				Spec: mellanoxv1alpha1.NicClusterPolicySpec{
+					OFEDDriver: &mellanoxv1alpha1.OFEDDriverSpec{
+						ForcePrecompiled: true,
+						ImageSpec: mellanoxv1alpha1.ImageSpec{
+							Image:            "mofed",
+							Repository:       "nvcr.io/nvidia/mellanox",
```

Comment:
The test is depends on the facts that NGC is available and that tag for Kernel generic-9.0.1 is not exist? Right?
I don't like the fact that we depend on uncontrolled external state in tests. Maybe we can mock DocaDriverImagesProvider? I think it is fine to create single mocked instance of DocaDriverImagesProvider which will work for entire tests suite. WDYT?

Reply:
Done